### PR TITLE
Move unit tests to net5.0

### PR DESCRIPTION
**Pull request by jaredpar**
This moves all of the unit tests tests to target `net5.0`.

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure @dotnet/roslyn-ide for review

**Comment by sharwell** [comment]
Why are we removing the netcoreapp3.1 testing?

**Comment by jaredpar** [comment]
> Why are we removing the netcoreapp3.1 testing?

This is generally what we do when moving from one .NET core to the next. Going forward the compiler in particular is only deployed on `net5.0` hence there isn't much reason to test there.

File: src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <RootNamespace>Microsoft.CodeAnalysis.UnitTests</RootNamespace>
-    <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
+    <TargetFrameworks>net5.0;net472</TargetFrameworks>
```
Comment (genlu) [inline src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj:7]: IDE will likely be on 3.1 until Net 6 in available though.

Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj:7]: Okay so you all want to stay on `netcoreapp3.1` for your testing?

Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj:7]: Note: at the moment you are not testing on `netcoreapp3.1`. You are testing on `net5.0`. The current setup for our branch just means you compile fro `netcoreapp3.1`

Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj]: Okay backed this out and kept the `netcoreapp3.1` portion in addition to `net5.0`

File: src/Compilers/Core/MSBuildTaskTests/TargetTests.cs [hunk 2]
```diff
@@ -265,15 +266,15 @@ public void GenerateEditorConfigCoreEvaluatesMetadata()
             Assert.Equal("_GeneratedEditorConfigMetadata", item.ItemType);
 
             var itemType = item.Metadata.SingleOrDefault(m => m.Name == "ItemType");
-            Assert.NotNull(itemType);
+            AssertEx.NotNull(itemType);
```
Comment (jcouv) [inline src/Compilers/Core/MSBuildTaskTests/TargetTests.cs:269]: just curious, why these changes?

File: src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj [hunk 3]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <RootNamespace>Microsoft.CodeAnalysis.UnitTests</RootNamespace>
-    <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
+    <TargetFrameworks>net5.0;netcoreapp3.1;net472</TargetFrameworks>
```
Comment (jcouv) [inline src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj:7]: I didn't understand why some projects add "net5.0" (keeping "netcoreapp3.1"), but others replace "netcoreapp3.1". What's the rationale?

File: eng/build.ps1 [hunk 4]
```diff
@@ -212,7 +212,7 @@ function BuildSolution() {
   $projects = Join-Path $RepoRoot $solution
   $toolsetBuildProj = InitializeToolset
 
-  $testTargetFrameworks = if ($testCoreClr) { "netcoreapp3.1" } else { "" }
+  $testTargetFrameworks = if ($testCoreClr) { 'net5.0%3Bnetcoreapp3.1' } else { "" }
```
Comment (jcouv) [inline eng/build.ps1:215]: Why not just use "net5.0"?

File: src/Compilers/Core/MSBuildTaskTests/TargetTests.cs (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/Core/MSBuildTaskTests/TargetTests.cs:269]: The new `net5.0` APIs brought in more nullable annotations and it made `SingleOrDefault` return a nullable value. The `Assert.NotNull` class doesn't have the `[NotNull]` attribute hence it was causing the line just below this to give a potential null ref warning. I added a helper that has the attribute so that we don't get the warning here. 

Note: I don't believe our upcoming compiler change will really help here because these APIs are still oblivious.

File: src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj (hunk 3 above)
Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/Roslyn.Services.UnitTests.Utilities.csproj:7]: All of the projects which replaced `netcoreapp3.1` with `net5.0` were unit test projects. The change reflected the desire to compile and run the tests against `net5.0`. This particular assembly is a utility assembly and used by compiler and IDE unit tests. 

As of this PR there are no IDE unit tests that target `netcoreapp3.1` however there are PRs to enable exactly this. Leaving the `netcoreapp3.1` here to support those PRs.

File: eng/build.ps1 (hunk 4 above)
Comment (jaredpar) [reply eng/build.ps1:215]: Similar to the above. The IDE is on the cusp of adding `netcoreapp3.1` tests into the repository. Given I'd already figured out the magic spell to run both TF correctly I kept it in here. Both of these changes were kept after some offline discussions with IDE (needed to understand their direction a bit better).
